tools/state-viewer/src/commands.rs

```diff
@@ -124,9 +125,12 @@ pub(crate) fn dump_tx(
         near_config.genesis.config.genesis_height,
         !near_config.client_config.archive,
     );
-    let hash = chain_store.get_block_hash_by_height(height)?;
-    let block = chain_store.get_block(&hash)?;
-    let txs = tx_dump(&mut chain_store, &block, select_account_ids);
+    let mut txs = vec![];
+    for height in start_height..=end_height {
+        let hash = chain_store.get_block_hash_by_height(height)?;
```

Comment:
what happens if one height is missing ? ( I guess in such case, the whole thing fails ? )